discussion_forum/tests.py

```diff
@@ -0,0 +1,592 @@
+        discussion_thread_id = 1
+        self.client.login(**ins_cred)
+        self._retrieve_discussion_thread_helper(discussion_thread_id)
+        self.client.logout()
+
+    def _create_discussion_thread_helper(self, title, status_code):
+        """helper function to test create discussion thread functionality.
+
+        Args:
+        title (str): title of the discussion thread
+        status_code (int): expected status code of the API call
+        """
+
+        data = {
+            "discussion_forum": 1,
+            "title": title,
+            "mark_as_important": True,
+            "author": 1,
+            "author_category": "I",
+            "description": "Description of discussion thread",
+            "pinned": True,
+            "anonymous_to_student": True,
+            "upvote": 0,
+            "downvote": 0,
+        }
+        url = reverse(
+            "discussion_forum:discussionthread-create-discussion-thread", args=[1]
+        )
+        response = self.client.post(url, data)
+        self.assertEqual(response.status_code, status_code)
+        if status_code == status.HTTP_201_CREATED:
+            return_data = response.data
+            for k in [
+                "created_on",
+                "modified_on",
+                "id",
+                "tag",
+            ]:
+                return_data.pop(k)
+            self.assertEqual(return_data, data)
+
+    def test_create_discussion_thread(self):
+        """
+        test to check: create a discussion thread.
+        """
+        self.client.login(**ins_cred)
+        self._create_discussion_thread_helper("Thread1", status.HTTP_201_CREATED)
+        self.client.logout()
+
+    def _update_discussion_thread_helper(self, title, status_code):
+        """
+        Helper function to test update discussion thread functionality.
+
+        Args:
+        status_code (int): expected status code of the API call
+        title (str): title of the discussion thread
+        """
+
+        discussion_thread1 = DiscussionThread(
+            title="DiscussionThread77",
+            discussion_forum_id=1,
+            author_id=1,
+            author_category="I",
+        )
+        discussion_thread1.save()
+        data = {
+            "discussion_forum": 1,
+            "title": title,
+            "mark_as_important": True,
+            "author": 1,
+            "author_category": "I",
+            "description": "Description of discussion thread",
+            "pinned": True,
+            "anonymous_to_student": True,
+            "upvote": 0,
+            "downvote": 0,
+        }
+        url = reverse(
+            ("discussion_forum:discussionthread-update-discussion-thread"),
+            kwargs={"pk": discussion_thread1.id},
+        )
+        response = self.client.put(url, data)
+        self.assertEqual(response.status_code, status_code)
+        if status_code == status.HTTP_200_OK:
+            return_data = response.data
+            for k in [
```

Comment:
same as create change proper var name for k and return_data

Reply:
done